# Homework

---

## Exercises

**Note**: write the program as clearly as possible, so that the user knows exactly what to do and another programmer will understand the code quickly. This goes for all exercises.

### Exercise 1 (1 point)

Ask a user to input two numbers. Is it true that their sum and product are the same?

#### Examples:
- `2 + 2 == 2 * 2` => `True`
- `3 + 3 == 3 * 3` => `False`

In [14]:
# Solution
# Dodeluvanje na boroevi
x = int(input('Enter the first number: '))
y = int(input('Enter the second number: '))

# Check if the sum and multiplication of two numbers istis equal
# if they are equal print True if not print False
if(x + y == x * y):
    print(True)
else:
    print(False)


Enter the first number:  3
Enter the second number:  3


False


---

### Exercise 2 (1 point)

Ask a user to enter any text of letters, integers, etc. Then, count the number of letters and integers that are in the text.
    
#### Examples:
- `text 1` => `# of letters: 4, # of integers 1`
- `text ` => `# of letters: 4, # of integers 0`

In [3]:
# Solution
string = input('')
digit = 0
alpha = 0

# ja napraviv so koristenje na vgradenite funkcii isdigit() i isalpha()
# isdigit proveruva dali ima cifra vo stringot
# isalpha proveruva dali ima karakteri vo stringot

for s in string:
    if(s.isdigit()):
        digit += 1
    if(s.isalpha()):
        alpha += 1
        
print(f'# of letter: {alpha}, # of integers: {digit}')




 Teodora 123


# of letter: 7, # of integers: 3


### Exercise 3 (1.5 points)

Enter an integer. Write a program to count the total number of digits in the integer. You should not use the built-in function `len()`
    
#### Examples:
- `478281`=> `# of digits: 6`
- `123` => `# of digits: 3`

In [41]:
# Solution
n = int(input('Please enter number: '))

# Write your code below
countDigits = 0

for el in str(n):
    countDigits += 1

print(f'# of digits: {countDigits}')

Please enter number:  12314


# of digits: 5


---

### Exercise 4 (1.5 point)

Ask for a user to enter any text. Then reverse the order of the text and show the result. 
    
#### Examples:
- `text` => `txet`
- `Brainster` => `retsniarB`

In [43]:
#Solution
string = input('Enter random text: ')

print(string[::-1])



Enter random text:  Brainster


retsniarB


---

### Exercise 5 (5 points)
Given the info about students (see the code block below), answer the following questions:
- What is the average number of points that all students received for the projects? (1 point)
- What is the average number of points per student? (1.5 point)
- Who is the best student i.e the student with the highest average points? (2.5 points)

In [9]:
# Solution
students = [{"student": "John", "course": "Computer Science",
             "projects": [{"Project 1": 93}, {"Project 2": 70}, {"Project 3": 70}, {"Project 4": 56},
                          {"Project 5": 40}]}, 
            
            {"student": "Shelly", "course": "Computer Science",
             "projects": [{"Project 1": 73}, {"Project 2": 94}, {"Project 3": 99},
                          {"Project 4": 45}, {"Project 5": 64}]},
            
            {"student": "Pat", "course": "Computer Science",
             "projects": [{"Project 1": 94}, {"Project 2": 50}, {"Project 3": 80}, {"Project 4": 44},
                          {"Project 5": 88}]}, 
            
            {"student": "Sam", "course": "Computer Science",
             "projects": [{"Project 1": 76}, {"Project 2": 95}, {"Project 3": 63},
                          {"Project 4": 48}, {"Project 5": 91}]},
            
            {"student": "Jeffrey", "course": "Computer Science",
             "projects": [{"Project 1": 96}, {"Project 2": 75}, {"Project 3": 100}, {"Project 4": 45},
                          {"Project 5": 75}]}, 
            
            {"student": "Irene", "course": "Computer Science",
             "projects": [{"Project 1": 86}, {"Project 2": 65}, {"Project 3": 88},
                          {"Project 4": 95}, {"Project 5": 88}]},
            
            {"student": "Mary", "course": "Computer Science",
             "projects": [{"Project 1": 65}, {"Project 2": 94}, {"Project 3": 43}, {"Project 4": 94},
                          {"Project 5": 45}]}, 
            
            {"student": "Molly", "course": "Computer Science",
             "projects": [{"Project 1": 68}, {"Project 2": 98}, {"Project 3": 82},
                          {"Project 4": 55}, {"Project 5": 60}]},
            
            {"student": "Beth", "course": "Computer Science",
             "projects": [{"Project 1": 64}, {"Project 2": 94}, {"Project 3": 62}, {"Project 4": 77},
                          {"Project 5": 92}]}, 
            
            {"student": "Jonathan", "course": "Computer Science",
             "projects": [{"Project 1": 100}, {"Project 2": 66}, {"Project 3": 55},
                          {"Project 4": 45}, {"Project 5": 42}]}]



students1 = [{"student": "John", "course": "Computer Science",
               "projects": [{"Project 1": 93}, {"Project 2": 70}, {"Project 3": 70}, {"Project 4": 56},
                          {"Project 5": 40}]},
            {"student": "Jonathan", "course": "Computer Science",
             "projects": [{"Project 1": 100}, {"Project 2": 66}, {"Project 3": 55},
                          {"Project 4": 45}, {"Project 5": 42}]}]

# Write your code below

sumStudents = 0
maxAvg = 0
maxIme = ''

# First I am making iteration in the list
for st in students:
    brojac = 0
    ime = ''
    course = ''
    # Here I accessed all the values from the given dictionary
    for v in st.values():
        
        # Here I made a counter so I can accsess the indexes from the values of the dicitnary
        brojac += 1
        
        # In the first if I am filling the value ime with the name of the student seperately
        if brojac == 1:
            ime = v
        
        # In the second if I am filling the course that they are studying
        if brojac == 2:
            course = v
            
        # In the third if I am accessing the list with dicitnaries inside
        if(brojac == 3):
            suma = 0
            avg = 0
            # Here I am counting the average score perstudent
            for l in v:
                l = list(l.values())
                suma += l[0]
            avg = suma/5
        # Here I am typing the name, course and score for each student
    print(f'The average points for {ime} that studies {course} is: {avg}')
    
    # Here I am counting the sum of all points of the students
    sumStudents += avg
    
     # Here I am finding the studing with the most points
    if(avg > maxAvg):
        maxAvg = avg
        maxIme = ime
        

            
print()
# Here I am writing the average score for all students
print(f'Vkupniot prosek sto go osovile site studenti zaedno e: {sumStudents/10}')


# Here I am writing the student's name and points that has the max points
print(f'The best student is  {maxIme} with: {maxAvg} points')


The average points for John that studies Computer Science is: 65.8
The average points for Shelly that studies Computer Science is: 75.0
The average points for Pat that studies Computer Science is: 71.2
The average points for Sam that studies Computer Science is: 74.6
The average points for Jeffrey that studies Computer Science is: 78.2
The average points for Irene that studies Computer Science is: 84.4
The average points for Mary that studies Computer Science is: 68.2
The average points for Molly that studies Computer Science is: 72.6
The average points for Beth that studies Computer Science is: 77.8
The average points for Jonathan that studies Computer Science is: 61.6

Vkupniot prosek sto go osovile site studenti zaedno e: 72.94000000000001
The best student is  Irene with: 84.4 points
